main.go

```diff
@@ -353,135 +356,124 @@ func _main(ctx context.Context) error {
+	blockHash, nodeHeight, err := dcrdClient.GetBestBlock()
+	if err != nil {
+		return fmt.Errorf("Unable to get block from node: %v", err)
+	}
 
-		// Since mining a block take approximately ChainParams.TargetTimePerBlock then the
-		// expected height of the best block from dcrd now should be this.
-		expectedHeight := int64(bestBlockAge/float64(activeChain.TargetTimePerBlock)) + nodeHeight
+	block, err := dcrdClient.GetBlockHeader(blockHash)
+	if err != nil {
+		return fmt.Errorf("unable to fetch the block from the node: %v", err)
+	}
 
-		// Estimate how far auxDB is behind the node.
-		blocksBehind := expectedHeight - lastBlockPG
-		if blocksBehind < 0 {
-			return fmt.Errorf("Node is still syncing. Node height = %d, "+
-				"DB height = %d", expectedHeight, heightDB)
-		}
+	// bestBlockAge is the time since the dcrd best block was mined.
+	bestBlockAge := time.Since(block.Timestamp).Minutes()
 
-		// PG gets winning tickets out of baseDB's pool info cache, so it must
-		// be big enough to hold the needed blocks' info, and charged with the
-		// data from disk. The cache is updated on each block connect.
-		tpcSize := int(blocksBehind) + 200
-		log.Debugf("Setting ticket pool cache capacity to %d blocks", tpcSize)
-		err = baseDB.GetStakeDB().SetPoolCacheCapacity(tpcSize)
-		if err != nil {
-			return err
-		}
+	// Since mining a block take approximately ChainParams.TargetTimePerBlock then the
+	// expected height of the best block from dcrd now should be this.
+	expectedHeight := int64(bestBlockAge/float64(activeChain.TargetTimePerBlock)) + nodeHeight
 
-		// Charge stakedb pool info cache, including previous PG blocks, up to
-		// best in sqlite.
-		if err = baseDB.ChargePoolInfoCache(heightDB - 2); err != nil {
-			return fmt.Errorf("Failed to charge pool info cache: %v", err)
-		}
+	// Estimate how far pgDB is behind the node.
+	blocksBehind := expectedHeight - lastBlockPG
+	if blocksBehind < 0 {
+		return fmt.Errorf("Node is still syncing. Node height = %d, "+
+			"DB height = %d", expectedHeight, heightDB)
+	}
 
-		// Fetch the latest blockchain info, which is needed to update the
-		// agendas db while db sync is in progress.
-		bci, err := baseDB.BlockchainInfo()
-		if err != nil {
-			return fmt.Errorf("failed to fetch the latest blockchain info")
-		}
+	// PG gets winning tickets out of baseDB's pool info cache, so it must
+	// be big enough to hold the needed blocks' info, and charged with the
+	// data from disk. The cache is updated on each block connect.
+	tpcSize := int(blocksBehind) + 200
+	log.Debugf("Setting ticket pool cache capacity to %d blocks", tpcSize)
+	err = stakeDB.SetPoolCacheCapacity(tpcSize)
+	if err != nil {
+		return err
+	}
 
-		// Update the current chain state in the ChainDBRPC
-		auxDB.UpdateChainState(bci)
+	// Charge stakedb pool info cache, including previous PG blocks, up to
+	// best in sqlite.
+	if err = baseDB.ChargePoolInfoCache(heightDB - 2); err != nil {
```

Comment:
`ChargePoolInfoCache` is a `WiredDB` method even though it is storing data in the `StakeDatabase` since it gets the data from the `WiredDB` backend.